File: src/main/java/io/pageobject/generator/locator/protractor/CssLocatorSource.java
package io.pageobject.generator.locator.protractor;

import io.pageobject.generator.GeneratorContext;
import io.pageobject.generator.locator.AbstractLocatorSource;
import org.jsoup.nodes.Element;

import static io.pageobject.generator.element.Elements.generateCssSelector;

public class CssLocatorSource extends AbstractLocatorSource {

    @Override
    public String extractLocatorValue(Element element, GeneratorContext context) {
        return generateCssSelector(element, context);
    }

    public String locatorTemplate() {
        return "by.css('%s')";
    }

}

File: lib/locator/protractor/css-locator-source.js
"use strict";

const AbstractLocatorSource = require('../abstract-locator-source');
const _ = require('lodash');
const generateCssSelector = require('../../element/elements').generateCssSelector;

const template = _.template("by.css('<%= value %>')");

class CssLocatorSource extends AbstractLocatorSource {

  extractLocatorValue(context) {
    return generateCssSelector(context.element, context);
  }

  locatorTemplate() {
    return template;
  }

}

module.exports = CssLocatorSource;